packages/pages/src/common/src/template/hydration.ts

```diff
@@ -134,21 +135,29 @@ const getCommonInjectedIndexHtml = (
  * @param clientHydrationString
  * @param indexHtml
  * @param appLanguage
+ * @param templatesConfig
  * @param headConfig
  * @returns the server template to render in the Vite dev environment
  */
 export const getIndexTemplateDev = (
   clientHydrationString: string | undefined,
   indexHtml: string,
   appLanguage: string,
+  templatesConfig: FeaturesConfig,
   headConfig?: HeadConfig
 ): string => {
-  return getCommonInjectedIndexHtml(
+  let commonIndex = getCommonInjectedIndexHtml(
     clientHydrationString,
     indexHtml,
     appLanguage,
     headConfig
   );
+  commonIndex = injectIntoEndOfHead(
+    commonIndex,
+    `<script id="templatesJsonDev">${JSON.stringify(templatesConfig, null)}</script>`
```

Comment:
Don't use an id and set the json object to a `const pageSets`

Reply:
Done